ShareX/TaskHelpers.cs

```diff
@@ -618,6 +618,10 @@ public static void AnnotateImage(string filePath, TaskSettings taskSettings = nu
                     AnnotateImageUsingGreenshot(null, filePath);
                 }
             }
+            else
+            {
+                AnnotateImage(taskSettings);
```

Comment:
This will make loop if dialog is canceled. Not to mention when someone right click to edit image that person won't like getting dialog to select random image.